provider/kubernetes/client.go

```diff
@@ -113,13 +116,32 @@ func createClientFromConfig(c *rest.Config) (*clientImpl, error) {
 // WatchAll starts namespace-specific controllers for all relevant kinds.
 func (c *clientImpl) WatchAll(namespaces Namespaces, stopCh <-chan struct{}) (<-chan interface{}, error) {
 	eventCh := make(chan interface{}, 1)
+	eventHandler := c.newResourceEventHandler(eventCh)
+
+	nsf := informers.NewSharedInformerFactory(c.clientset, resyncPeriod)
+	nsf.Core().V1().Namespaces().Informer().AddEventHandler(eventHandler)
+	c.namespaceFactory = nsf
```

Comment:
is there any reason for doing that over the following ?

```go
c.namespaceFactory = informers.NewSharedInformerFactory(c.clientset, resyncPeriod)
c.namespaceFactory.Core().V1().Namespaces().Informer().AddEventHandler(eventHandler)
```

Reply:
`non-name c.namespaceFactory on left side of :=` ;)

Reply:
oooo cleaner!